express_relay/sdk/python/express_relay/svm/limo_client.py

```diff
@@ -0,0 +1,337 @@
+        return account_info.value is not None
+
+    def create_associated_token_account_idempotent(
+        self, payer: Pubkey, owner: Pubkey, mint: Pubkey, token_program_id: Pubkey
+    ) -> Instruction:
+        """Creates a transaction instruction to create an associated token account.
+
+        Returns:
+            The instruction to create the associated token account.
+        """
+        associated_token_address = self.get_ata(owner, mint, token_program_id)
+        return Instruction(
+            accounts=[
+                AccountMeta(pubkey=payer, is_signer=True, is_writable=True),
+                AccountMeta(
+                    pubkey=associated_token_address, is_signer=False, is_writable=True
+                ),
+                AccountMeta(pubkey=owner, is_signer=False, is_writable=False),
+                AccountMeta(pubkey=mint, is_signer=False, is_writable=False),
+                AccountMeta(
+                    pubkey=SYSTEM_PROGRAM_ID, is_signer=False, is_writable=False
+                ),
+                AccountMeta(
+                    pubkey=token_program_id, is_signer=False, is_writable=False
+                ),
+                AccountMeta(pubkey=RENT, is_signer=False, is_writable=False),
+            ],
+            program_id=ASSOCIATED_TOKEN_PROGRAM_ID,
+            data=bytes(1),  # idempotent version of the instruction
+        )
+
+    async def get_ata_and_create_ixn_if_required(
+        self,
+        owner: Pubkey,
+        token_mint_address: Pubkey,
+        token_program_id: Pubkey,
+        payer: Pubkey,
+    ) -> Tuple[Pubkey, Sequence[Instruction]]:
+        ata = self.get_ata(owner, token_mint_address, token_program_id)
+        if not await self.account_exists(ata):
+            ix = self.create_associated_token_account_idempotent(
+                payer, owner, token_mint_address, token_program_id
+            )
+            return ata, [ix]
+        return ata, []
+
+    async def get_init_if_needed_wsol_create_and_close_ixs(
+        self, owner: Pubkey, payer: Pubkey, amount_to_deposit_lamports: int
+    ) -> WSOLInstructions:
+        """
+        Returns necessary instructions to create, fill and close a wrapped SOL account.
+        If the account already exists:
+            it makes sure the balance is at least `amount_to_deposit_lamports` with no create or close instructions.
+        If the account does not exist:
+            it creates the account, fills it with `amount_to_deposit_lamports` lamports and close it in the end.
+        Args:
+            owner: Who owns the WSOL token account
+            payer: Who pays for the instructions
+            amount_to_deposit_lamports: Minimum amount of lamports required in the account
+        """
+        ata = self.get_ata(owner, WRAPPED_SOL_MINT, TOKEN_PROGRAM_ID)
+        ata_info = await self._connection.get_account_info(ata)
+        current_balance = 0
+        create_ixs = []
```

Comment:
```
create_ixs = [] if ata_info.value else  [
                self.create_associated_token_account_idempotent(
                    payer, owner, WRAPPED_SOL_MINT, TOKEN_PROGRAM_ID
                )
            ]
```

Reply:
```
current_balance = TOKEN_ACCOUNT_LAYOUT.parse(ata_info.value.data).amount if ata_info.value else 0
```

Reply:
The reason I'm suggesting this is that created_ixs and current_balance is coupled in the code and I think they shouldn't be :-? (Not sure btw)

Reply:
+ You can couple the close_ixs with create_ixs by using tuple in the inline if statement

Reply:
the inline if doesn't look great on these big expressions, so I'd rather keep it like this for the big ones. I tried to move stuff around to make it more readable